### PR TITLE
extend ziti_client_cb API for better receiving of peer supplied data

### PR DESCRIPTION
**Pull request by ekoby**

### CODE REVIEW

File: programs/sample-host/sample-host.c [hunk 1]
```diff
@@ -58,17 +58,17 @@ static void on_client_connect(ziti_connection clt, int status) {
     }
 }
 
-static void on_client(ziti_connection serv, ziti_connection client, int status) {
+static void on_client(ziti_connection serv, ziti_connection client, int status, ziti_client_ctx *clt_ctx) {
     if (status == ZITI_OK) {
         const char *source_identity = ziti_conn_source_identity(client);
```
Comment (scareything) [inline programs/sample-host/sample-host.c:63]: should this sample show using ->caller_id?

File: includes/ziti/ziti.h [hunk 2]
```diff
@@ -330,10 +336,11 @@ typedef void (*ziti_conn_cb)(ziti_connection conn, int status);
  * @param serv hosting connection, initialized with ziti_listen()
  * @param client client connection - generally passed to ziti_accept() in this function
  * @param status #ZITI_OK or error
+ * @param ctx object containing application data passed by dialing identity, \see ziti_dial_opts
```
Comment (scareything) [inline includes/ziti/ziti.h:339]: it may be obvious, but should this mention that the contents of `ctx` are only valid in the client_cb?

**Review by scareything — APPROVED** [review]

File: programs/sample-host/sample-host.c (hunk 1 above)
Comment (ekoby) [reply programs/sample-host/sample-host.c]: done

File: includes/ziti/ziti.h (hunk 2 above)
Comment (ekoby) [reply includes/ziti/ziti.h]: done

**Pull request merged**